### ISSUE TITLE
GKE Cluster Not Logging Properly

### ISSUE THREAD

**Issue by pnovotnak** [kubernetes/kubernetes#20516]
The company I work for has a single-node cluster running which has had DNS issues (names won't resolve) crop up twice over the past couple weeks. In both cases the solution was to recycle the VM. In the process of debugging this problem, I ran into another... Our cluster doesn't seem to be logging properly to the Compute Engine Console, even though our cluster has Monitoring.full and Logging.write access. Here's some additional information;

Visiting https://console.developers.google.com/logs?... in a browser;
![console-logs](https://cloud.githubusercontent.com/assets/5713245/12765088/be9b3c70-c9b1-11e5-98d0-2d62384c0c6e.png)

```
$ kubectl logs fluentd-cloud-logging-gke-...

2016-02-02 20:33:03 +0000 [warn]: Dropping 70 log message(s) error_class="Google::APIClient::ClientError" error="Project has not enabled the API. Please use Google Developers Console to activate the 'logging' API for your project."
2016-02-02 20:33:03 +0000 [warn]: Dropping 10 log message(s) error_class="Google::APIClient::ClientError" error="Project has not enabled the API. Please use Google Developers Console to activate the 'logging' API for your project."
2016-02-02 20:33:03 +0000 [warn]: Dropping 3 log message(s) error_class="Google::APIClient::ClientError" error="Project has not enabled the API. Please use Google Developers Console to activate the 'logging' API for your project."
2016-02-02 20:33:03 +0000 [warn]: Dropping 1 log message(s) error_class="Google::APIClient::ClientError" error="Project has not enabled the API. Please use Google Developers Console to activate the 'logging' API for your project."
2016-02-02 20:33:03 +0000 [warn]: Dropping 1 log message(s) error_class="Google::APIClient::ClientError" error="Project has not enabled the API. Please use Google Developers Console to activate the 'logging' API for your project."
2016-02-02 20:33:04 +0000 [warn]: Dropping 2 log message(s) error_class="Google::APIClient::ClientError" error="Project has not enabled the API. Please use Google Developers Console to activate the 'logging' API for your project."
2016-02-02 20:33:04 +0000 [warn]: Dropping 3 log message(s) error_class="Google::APIClient::ClientError" error="Project has not enabled the API. Please use Google Developers Console to activate the 'logging' API for your project."
2016-02-02 20:33:04 +0000 [warn]: Dropping 1 log message(s) error_class="Google::APIClient::ClientError" error="Project has not enabled the API. Please use Google Developers Console to activate the 'logging' API for your project."
2016-02-02 20:33:04 +0000 [warn]: Dropping 2 log message(s) error_class="Google::APIClient::ClientError" error="Project has not enabled the API. Please use Google Developers Console to activate the 'logging' API for your project."
2016-02-02 20:33:04 +0000 [warn]: Dropping 1 log message(s) error_class="Google::APIClient::ClientError" error="Project has not enabled the API. Please use Google Developers Console to activate the 'logging' API for your project."
2016-02-02 20:33:04 +0000 [warn]: Dropping 1 log message(s) error_class="Google::APIClient::ClientError" error="Project has not enabled the API. Please use Google Developers Console to activate the 'logging' API for your project."
2016-02-02 20:33:04 +0000 [warn]: Dropping 1 log message(s) error_class="Google::APIClient::ClientError" error="Project has not enabled the API. Please use Google Developers Console to activate the 'logging' API for your project."
2016-02-02 20:33:04 +0000 [warn]: Dropping 3 log message(s) error_class="Google::APIClient::ClientError" error="Project has not enabled the API. Please use Google Developers Console to activate the 'logging' API for your project."
```

Not sure how to fix this. I saw #15727, but it seems like the solution for that issue is already in place here.

**Comment by jlowdermilk** [comment]
Please re-post your question to [stackoverflow](http://stackoverflow.com/questions/tagged/kubernetes). 

We are trying to consolidate the channels to which questions for help/support are posted so that we can improve our efficiency in responding to your requests, and to make it easier for you to find answers to frequently asked questions and how to address common use cases. 

We regularly see messages posted in multiple forums, with the full response thread only in one place or, worse, spread across multiple forums. Also, the large volume of support issues on github is making it difficult for us to use issues to identify real bugs.

The Kubernetes team scans stackoverflow on a regular basis, and will try to ensure your questions don't go unanswered.

Before posting a new question, please search stackoverflow for answers to similar questions, and also familiarize yourself with:
   * [the user guide](http://kubernetes.io/v1.0/)
   * [the troubleshooting guide](http://kubernetes.io/v1.0/docs/troubleshooting.html)

Again, thanks for using Kubernetes.

The Kubernetes Team

**Comment by pires** [comment]
@pnovotnak `Please use Google Developers Console to activate the 'logging' API for your project."` - you need to enable `Logging API` for the project where this cluster was created.

Also, make sure your GKE is 1.1.7 since 1.1.4 had issues with logging.

**Comment by pnovotnak** [comment]
This issue was solved exactly how the error message suggested... Enabling the API in the console. I didn't want to create additional traffic on this ticket because it truly is an operator/ platform issue, not a kubernetes one, but here's the stackoverflow post with solution for reference;

http://stackoverflow.com/questions/35165973/fluentd-dropping-messages

TL;DR

"You can do so from the Developers Console, [here](https://console.developers.google.com/apis/api/logging/overview). Try going there, clicking the Enable API button, and seeing whether the errors keep coming.
